setup.py

```diff
@@ -350,9 +350,9 @@ def build(self):
         # Fetch all c and cpp files from the subdirs to compile.
         for library_name in self._sub_library_names:
             for extension in ["*.c", "*.cpp"]:
-                self._source_files.extend(glob.glob(
+                self._source_files.extend(sorted(glob.glob(
```

Comment:
When writing the patch I found it most straight-forward to get rid of the randomness right after it is added here by glob.
I tested that `sorted(self._source_files)` in line 357 also does the trick. Feel free to merge either variant.

I only tested the build, not if the result still works.
And I did not understand your 2nd question.